paddle/fluid/operators/repeat_interleave_op.cu

```diff
@@ -0,0 +1,307 @@
+
+    int dim = context.Attr<int>("dim");
+    auto input_dim = in_grad->dims();
+    auto output_dim = output_grad->dims();
+    dim = dim >= 0 ? dim : dim + input_dim.size();
+    auto stride_dim = framework::stride(input_dim);
+    int64_t stride = stride_dim[dim];
+    int64_t size = output_dim[dim];
+    int64_t delta = input_dim[dim] - size;
+
+    int64_t numel = in_grad->numel();
+    int64_t out_nums = output_grad->numel();
+
+    auto stream =
+        context.template device_context<platform::CUDADeviceContext>().stream();
+
+    index_select_grad_init<
+        T><<<(numel + PADDLE_CUDA_NUM_THREADS - 1) / PADDLE_CUDA_NUM_THREADS,
+             PADDLE_CUDA_NUM_THREADS, 0, stream>>>(in_grad_data, numel);
+
+    int repeats = context.Attr<int>("Repeats");
+    framework::LoDTensor index;
+    if (context.HasInput("RepeatsTensor")) {
+      auto repeats_tensor =
+          context.Input<framework::LoDTensor>("RepeatsTensor");
+
+      const auto& index_type = repeats_tensor->type();
+      bool index_type_match = index_type == framework::proto::VarType::INT64 ||
+                              index_type == framework::proto::VarType::INT32;
+      PADDLE_ENFORCE_EQ(
+          index_type_match, true,
+          platform::errors::InvalidArgument(
+              "Input(Index) holds the wrong type, it holds %s, but "
+              "desires to be %s or %s",
+              paddle::framework::DataTypeToString(index_type),
+              paddle::framework::DataTypeToString(
+                  framework::proto::VarType::INT32),
+              paddle::framework::DataTypeToString(
+                  framework::proto::VarType::INT64)));
+
+      if (index_type == framework::proto::VarType::INT64) {
+        RepeatsTensor2IndexTensor<DeviceContext, int64_t>(*repeats_tensor,
+                                                          &index);
+        int64_t index_nums = index.numel();
+
+        const int64_t* index_data = index.data<int64_t>();
+        index_select_grad_cuda_kernel<T, int64_t><<<
+            (out_nums + PADDLE_CUDA_NUM_THREADS - 1) / PADDLE_CUDA_NUM_THREADS,
+            PADDLE_CUDA_NUM_THREADS, 0, stream>>>(
+            output_grad_data, in_grad_data, index_data, index_nums, out_nums,
+            stride, size, delta);
+        platform::GpuStreamSync(stream);
+      } else {
+        RepeatsTensor2IndexTensor<DeviceContext, int>(*repeats_tensor, &index);
+        int64_t index_nums = index.numel();
+
+        const int* index_data = index.data<int>();
+        index_select_grad_cuda_kernel<T, int><<<
+            (out_nums + PADDLE_CUDA_NUM_THREADS - 1) / PADDLE_CUDA_NUM_THREADS,
+            PADDLE_CUDA_NUM_THREADS, 0, stream>>>(
+            output_grad_data, in_grad_data, index_data, index_nums, out_nums,
+            stride, size, delta);
+        platform::GpuStreamSync(stream);
+      }
+    } else if (repeats > 0) {
+      int index_size = in_grad->dims()[dim] * repeats;
```

Comment:
in_grad->dims()[dim] will return data type of int64_t, index_size should be define as int64_t ?